openvdb/openvdb/tools/Count.h

```diff
@@ -0,0 +1,83 @@
+// Copyright Contributors to the OpenVDB Project
+// SPDX-License-Identifier: MPL-2.0
+//
+/// @file   Count.h
```

Comment:
Maybe this should live in FindActiveValues? Or the two can be combined?

Reply:
This is a good question, I was thinking about combining them. Tagging @kmuseth for visibility.

I've just pushed up a change to rename `tools::activeVoxelCount` to `tools::countActiveVoxels` and migrated just one of the methods from FindActiveValues - the `tools::countActiveValues` method. I renamed that to `tools::countActiveVoxels` and rewrote it to use the DynamicNodeManager as well, so these are the two signatures:

```
template <typename TreeT>
Index64 countActiveVoxels(const TreeT& tree, bool threaded = true);
template <typename TreeT>
Index64 countActiveVoxels(const TreeT& tree, const CoordBBox& bbox, bool threaded = true);
```

I tested the performance of this new method against the FindActiveValues one for three different cases:

```
Entire BBox -  FindActiveValues 290ms  vs Count 28ms
Tiny BBox -    FindActiveValues 0.02ms vs Count 0.43ms
Partial BBox - FindActiveValues 59ms   vs Count 9.3ms
```

It's not entirely a fair test because I noticed a significant optimization for the test case I was using (a sphere level set) which is to handle the common case where the leaf node bounding box is entirely contained by the testing bounding box. If I apply that optimization to FindActiveValues, they're much closer:

```
Entire BBox - FindActiveValues 39ms vs Count 28ms
```

It doesn't do everything that the FindActiveValues header does, but it's a lot less code - I'm happy to see the DynamicNodeManager working well for this use case too.

Finally, I added `tools::memUsage` to this Count header and achieved a similar speed up there too:

`Tree::memUsage 184ms vs tools::memUsage 22ms`

I _think_ it makes sense to put this method here as it is counting the bytes of memory it occupies.